### ISSUE TITLE
fetchmail: mailbox selection failed

### ISSUE THREAD

**Issue by parazyd** [dyne/JaroMail#18]
When using mail.archlinux.info and imap.gmx.com, these errors appear when fetching emails.
```
 (*) Jaro Mail v4.0 running on GNU/Linux
  .  System in /usr/local/share/jaromail
 [D] Loading zlib: /usr/local/share/jaromail/zlibs/publish
 [D] Loading zlib: /usr/local/share/jaromail/zlibs/helpers
 [D] Loading zlib: /usr/local/share/jaromail/zlibs/stats
 [D] Loading zlib: /usr/local/share/jaromail/zlibs/keyring
 [D] Loading zlib: /usr/local/share/jaromail/zlibs/addressbook
 [D] Loading zlib: /usr/local/share/jaromail/zlibs/imap
 [D] Loading zlib: /usr/local/share/jaromail/zlibs/zuper
 [D] Loading zlib: /usr/local/share/jaromail/zlibs/filters
 [D] Loading zlib: /usr/local/share/jaromail/zlibs/maildirs
 [D] Loading zlib: /usr/local/share/jaromail/zlibs/search
 [D] Loading zlib: /usr/local/share/jaromail/zlibs/accounts
 [D] Loading zlib: /usr/local/share/jaromail/zlibs/parse
 [D] Loading zlib: /usr/local/share/jaromail/zlibs/locking
 [D] Loading zlib: /usr/local/share/jaromail/zlibs/email
 [D] Loading zlib: /usr/local/share/jaromail/zlibs/zuper.init
 [D] Zuper 0.2 initialized
 [D] 9 global variables registered
 [D] 6 global arrays registered
  .  full set of auxiliary functions loaded
 [D] Zuper 0.2 initialized
 [D] 9 global variables registered
 [D] 6 global arrays registered
 [D] Rm binary: wipe -f -s -q -R /dev/urandom
 [D] Mutt binary: /bin/mutt
 [D] Notmuch binary: /bin/notmuch
 [D] Alot binary: /bin/alot
 [D] All debug messages ON
 [D] fetch
 [D] read_account looking for archinfo.txt
 [D] name: xxx@archlinux.info 
 [D] email: xxx@archlinux.info
 [D] login: xxx@archlinux.info
 [D] host: 
 [D] port: 
 [D] imap: mail.archlinux.info
 [D] imap port: 993
 [D] smtp: mail.archlinux.info
 [D] smtp port: 465
 [D] trans: TLS1
 [D] certfile: 
 [D] cert: check
 [D] auth: plain
 [D] options: keep 
 [D] folders: 
 [D] exclude: 
 [D] nm_setup
  .  notmuch setup unread
 (*) Fetching email for account archinfo.txt
 [D] Test if we are online
 [D] trying to connect mail.archlinux.info port 993
  .  Host mail.archlinux.info responds on port 993
 [D] ask_password
  .  retrieving login credentials for xxx@archlinux.info (archinfo.txt)
  .  username 'xxx@archlinux.info' on host 'mail.archlinux.info'
 [D] keyring path: jaromail/xxx@archlinux.info/mail.archlinux.info
  .  looking for password in secret-tool
  .  saved password found in keyring
 [D] imap_list_folders()
 [D] run_imap_query
 [D] running tls imap query via openssl, encrypted
  .  5 folders found
 [D] fetch folders: / / / / INBOX
fetchmail: mailbox selection failed
fetchmail: client/server synchronization error while fetching from parazyd@archlinux.info@mail.archlinux.info
 [W] Mailbox selection failed (fetchmail: normal termination, status 7
  .  please wait while downloading mails to incoming...
fetchmail: mailbox selection failed
fetchmail: client/server synchronization error while fetching from parazyd@archlinux.info@mail.archlinux.info

fetchmail: Query status=7 (ERROR)

 [D] update_filters
 (*) Updating filters...
 [D] newlock /home/parazyd/Mail/cache/filters
 [D] lock /home/parazyd/Mail/cache/filters
 [D] insert filter rules in the cache
 [D] to: <announce@archlinux> -> zz.archann
 [D] maildir already exists: /home/parazyd/Mail/zz.archann
 [D] to: <arch-dev-public@archlinux> -> zz.archdev
 [D] maildir already exists: /home/parazyd/Mail/zz.archdev
 [D] to: <arch-general@archlinux> -> zz.archgen
 [D] maildir already exists: /home/parazyd/Mail/zz.archgen
 [D] to: <arch-projects@archlinux> -> zz.archproj
 [D] maildir already exists: /home/parazyd/Mail/zz.archproj
 [D] to: <arch-security@archlinux> -> zz.archsec
 [D] maildir already exists: /home/parazyd/Mail/zz.archsec
 [D] to: <bitcoin-dev@> -> zz.bitcoindev
 [D] maildir already exists: /home/parazyd/Mail/zz.bitcoindev
 [D] to: <twitter.com> -> zz.social
 [D] maildir already exists: /home/parazyd/Mail/zz.social
 [D] to: <linkedin.com> -> zz.social
 [D] maildir already exists: /home/parazyd/Mail/zz.social
 [D] to: <plus.google.com> -> zz.social
 [D] maildir already exists: /home/parazyd/Mail/zz.social
 [D] to: <youtube.com> -> zz.social
 [D] maildir already exists: /home/parazyd/Mail/zz.social
 [D] to: <wmt-noreply@google> -> zz.social
 [D] maildir already exists: /home/parazyd/Mail/zz.social
 [D] to: <facebook> -> zz.social
 [D] maildir already exists: /home/parazyd/Mail/zz.social
 [D] compile the list of own addresses and aliases
 [D] unlocking and compiling the cache
 [D] unlock /home/parazyd/Mail/cache/filters
 [D] pidcheck /home/parazyd/Mail/cache/filters
 [D] pidcheck: 14993
 [D] /home/parazyd/Mail/cache/filters lock belongs to us
 [D] recursive reload
 [D] filter_maildir
 (*) Filtering maildir: incoming (1 mails}
 [D] update_notmuch
 (*) Updating notmuch indexes
 [D] nm_setup
  .  notmuch setup 
 [D] notmuch new
 [D] lock /home/parazyd/Mail/cache/notmuch
No new mail.
 [D] unlock /home/parazyd/Mail/cache/notmuch
 [D] pidcheck /home/parazyd/Mail/cache/notmuch
 [D] pidcheck: 14993
 [D] /home/parazyd/Mail/cache/notmuch lock belongs to us
 [D] endgame EXIT
 [D] destructor: _ztmp_destructor
 [D] _ztmp_destructor
```
The script exits successfully and doesn't hang at any point.
Using:
* Arch Linux
* Latest Jaromail from git
* openssl 1.0.2.e-1
* fetchmail 6.3.26-3

**Comment by parazyd** [comment]
Worked around by manually specifying folders. Still would not close the issue because jaromail unsucessfully used the automatic folder detection.

**Comment by jaromil** [comment]
Just tested `jaro imap listfolders` which is a direct call to the function that discovers imap remote folders on servers: returns an empty array. So I think I can reproduce this.

**Comment by jaromil** [comment]
this was fixed in c177c8c3c830eceaf8ba7a4dee2fced7a3f8d776
